narwhals/expression.py

```diff
@@ -74,6 +74,54 @@ def cast(
         self,
         dtype: Any,
     ) -> Expr:
+        """
+        Redefine an object's data type.
+
+        Arguments:
+            dtype: Data type that the object will be cast into.
+
+        Examples:
+            >>> import pandas as pd
+            >>> import polars as pl
+            >>> import narwhals as nw
+            >>> from datetime import date
+            >>> df_pd = pd.DataFrame(
+            ... {"foo": [1, 2, 3],
+            ... "bar": [6.0, 7.0, 8.0],
+            ... "ham": [date(2020, 1, 2), date(2021, 3, 4), date(2022, 5, 6)],})
```

Comment:
`'ham'` doesn't really add too much to this example - perhaps we can remove it, so that the definitions of `df_pd` and `df_pl` can each fit on a single line?